src/sqs/sqs.types.ts

```diff
@@ -0,0 +1,66 @@
+import { WebhookPayloadDeploymentStatus } from "@octokit/webhooks";
+import type { WebhookPayloadCreate } from "@octokit/webhooks";
+import type { TaskType } from "~/src/sync/sync.types";
+
+export type BranchMessagePayload = {
+	jiraHost: string,
+	installationId: number,
+	gitHubAppConfig?: GitHubAppConfig, //undefined for cloud
+	webhookReceived: number,
+	webhookId: string,
+
+	// The original webhook payload from GitHub. We don't need to worry about the SQS size limit because metrics show
+	// that payload size for deployment_status webhooks maxes out at 9KB.
+	webhookPayload: WebhookPayloadCreate,
+}
+
+export type BackfillMessagePayload = {
+	jiraHost: string,
+	installationId: number,
+	gitHubAppConfig?: GitHubAppConfig, //undefined for cloud
+	startTime?: string,
+	commitsFromDate?: string,
+	targetTasks?: TaskType[]
+}
+
+export type DeploymentMessagePayload = {
+	jiraHost: string,
+	installationId: number,
+	gitHubAppConfig?: GitHubAppConfig, //undefined for cloud
+	webhookReceived: number,
+	webhookId: string,
+
+	// The original webhook payload from GitHub. We don't need to worry about the SQS size limit because metrics show
+	// that payload size for deployment_status webhooks maxes out at 13KB.
+	webhookPayload: WebhookPayloadDeploymentStatus,
+}
+
+export type PushQueueMessagePayload = {
+	jiraHost: string,
+	installationId: number,
+	gitHubAppConfig?: GitHubAppConfig, //undefined for cloud
+	repository: PayloadRepository,
+	shas: { id: string, issueKeys: string[] }[],
+	webhookId: string,
+	webhookReceived?: number,
+}
+
+type GitHubAppConfig = {
+	appId: number,
+	clientId: string,
+	gitHubBaseUrl: string,
+	uuid: string,
+	gitHubAppId: string,
+	//gitHubClientSecret: string,
```

Comment:
Hi @mboudreau and @thombergs , I've temporary commented this `secrets` in this PR. Let me think a bit more of it, and raise it in another PR.

----

For now, I need this PR as me and @Harminder84 is working on pushing and processing the queue. Need the interface change.